tests/unit_test/unit-test-ec2.py

```diff
@@ -0,0 +1,73 @@
+import boto3
+from moto import mock_ec2, settings
+import pytest
+from botocore.exceptions import ClientError
+import re
+from config import ec2_ami, region, ec2_az
+
+
+@mock_ec2
+def test_instance_launch_and_terminate():
+    client = boto3.client("ec2", region_name=region)
+
+    with pytest.raises(ClientError) as ex:
+        client.run_instances(ImageId=ec2_ami, MinCount=1, MaxCount=1, DryRun=True)
+    ex.value.response["ResponseMetadata"]["HTTPStatusCode"] == 412
+    ex.value.response["Error"]["Code"] == "DryRunOperation"
+    ex.value.response["Error"][
+        "Message"
+    ] == "An error occurred (DryRunOperation) when calling the RunInstances operation: Request would have succeeded, but DryRun flag is set"
+
+    reservation = client.run_instances(ImageId=ec2_ami, MinCount=1, MaxCount=1)
+    len(reservation["Instances"]) == 1
+    instance = reservation["Instances"][0]
+    instance["State"] == {"Code": 0, "Name": "pending"}
+    instance_id = instance["InstanceId"]
+
+    reservations = client.describe_instances(InstanceIds=[instance_id])["Reservations"]
+    len(reservations) == 1
+    reservations[0]["ReservationId"] == reservation["ReservationId"]
+    instances = reservations[0]["Instances"]
+    len(instances) == 1
```

Comment:
can we build logic dynamically instead of hardcoding values in if condition

Reply:
Resolved in [d67f4bc](https://github.com/XgridInc/xc3/pull/18/commits/d67f4bcc757b95ad043c435e8c4acb4dd33c6a76)